x-pack/plugins/apm/public/components/shared/Links/apm/AnomalyDetectionSetupLink.test.tsx

```diff
@@ -4,51 +4,88 @@
+  });
+
+  const { baseElement, container } = render(
+    <MissingJobsAlert environment={environment} />
+  );
+
+  // hover tooltip anchor if it exists
+  const toolTipAnchor = container.querySelector('.euiToolTipAnchor') as any;
+  if (toolTipAnchor) {
+    fireEvent.mouseOver(toolTipAnchor);
+
+    // wait for tooltip text to be in the DOM
+    await wait(() => {
+      const toolTipText = baseElement.querySelector('.euiToolTipPopover')
+        ?.textContent;
+      expect(toolTipText).not.toBe(undefined);
     });
-    it('should return false when environment is included in the jobs', () => {
-      const result = showAlert(dataWithJobs, 'staging');
-      expect(result).toBe(false);
+  }
+
+  const toolTipText = baseElement.querySelector('.euiToolTipPopover')
+    ?.textContent;
+
+  return { toolTipText, toolTipAnchor };
+}
+
+describe('MissingJobsAlert', () => {
+  it('when no jobs exists', async () => {
+    const { toolTipText, toolTipAnchor } = await renderTooltipAnchor({
+      jobs: [],
     });
+
+    expect(toolTipAnchor).toBeInTheDocument();
+    expect(toolTipText).toBe(
+      'Anomaly detection is not yet enabled. Click to continue setup.'
+    );
   });
 
-  describe('there is no environment selected (All)', () => {
-    it('should return true when there are no jobs', () => {
-      const result = showAlert(dataWithoutJobs, undefined);
-      expect(result).toBe(true);
+  it('when no job exists for the selected environment', async () => {
+    const { toolTipAnchor, toolTipText } = await renderTooltipAnchor({
+      jobs: [{ environment: 'production', job_id: 'my_job_id' }],
+      environment: 'staging',
     });
-    it('should return false when there are any number of jobs', () => {
-      const result = showAlert(dataWithJobs, undefined);
-      expect(result).toBe(false);
+
+    expect(toolTipAnchor).toBeInTheDocument();
+    expect(toolTipText).toBe(
+      'Anomaly detection is not yet enabled for the environment "staging". Click to continue setup.'
+    );
+  });
+
+  it('when a job exists for the selected environment', async () => {
+    const { toolTipAnchor, toolTipText } = await renderTooltipAnchor({
+      jobs: [{ environment: 'production', job_id: 'my_job_id' }],
+      environment: 'production',
     });
+
+    expect(toolTipAnchor).not.toBeInTheDocument();
+    expect(toolTipText).toBe(undefined);
   });
 
-  describe('when a known error occurred', () => {
-    it('should return false', () => {
-      const data = ({
-        errorCode: 'MISSING_READ_PRIVILEGES',
-      } as unknown) as any;
-      const result = showAlert(data, undefined);
-      expect(result).toBe(false);
+  it('when at least one job exists and no environment is selected', async () => {
+    const { toolTipAnchor, toolTipText } = await renderTooltipAnchor({
+      jobs: [{ environment: 'production', job_id: 'my_job_id' }],
     });
+
+    expect(toolTipAnchor).not.toBeInTheDocument();
+    expect(toolTipText).toBe(undefined);
   });
```

Comment:
I think the tests became very clean and easy to read. It did come at the cost of making `renderTooltipAnchor` slightly more complex. All in all I think it's worth it. LMK what you think

Reply:
It's a good compromise. It's unfortunate that we can't encapsulate all this logic into the original `showAlert` function, since that was simple to use & test. But then we'd be making extra API calls that might not be necessary, so this works well.